### PR TITLE
Draft out an implementation for #731

### PR DESCRIPTION
**Pull request by cpitclaudel**
Here's a draft implementation of the approach you suggested in #731 for discussion; it doesn't have docstrings yet. It's pretty straightforward, but computing the proper offset is more verbose than I would have hoped.

What do you think?

Connects to #731

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -3652,11 +3652,32 @@ Return a list with the contents of the table cell."
...
+                   (flycheck-flush-multiline-message message))
                   (flycheck-error-list-make-cell
                    (format "(%s)" checker)
                    'flycheck-error-list-checker-name)))))
 
+(defun flycheck-compute-message-column-offset ()
+  (let* ((widths (mapcar (lambda (fmt)
+                           (let* ((name (nth 0 fmt))
+                                  (width (nth 1 fmt))
+                                  (props (nthcdr 3 fmt))
```
Comment (lunaryorn) [inline flycheck.el]: Please use `pcase-let` to destructure `fmt`.

**Comment by cpitclaudel** [comment]
Thanks for the suggestion! `pcase-let` does make it nicer.

File: flycheck.el [hunk 2]
```diff
@@ -3652,11 +3652,34 @@ Return a list with the contents of the table cell."
...
+                                  widths)))
+    (apply #'+ tabulated-list-padding (mapcar #'cdr before-msg))))
+
+(defun flycheck-flush-multiline-message (msg)
+  "Prepare error message MSG for display in the error list.
+
+Prepend all lines of MSG except the first with enough space to
+ensure that they line up properly once the message is displayed."
+  (let* ((msg-offset (flycheck-compute-message-column-offset))
+         (spc (propertize " " 'display `(space . (:width ,msg-offset))))
```
Comment (lunaryorn) [inline flycheck.el]: Nice :)

**Comment by lunaryorn** [comment]
@cpitclaudel I pretty much like this implementation.  I'll try to test it this soon to look how it feels, and merge eventually.  Meanwhile, would you mind to update the commit message?  It's not a “draft” implementation anymore, after all ;)

**Comment by lunaryorn** [comment]
@cpitclaudel And, if you don't mind, please rebase on master and add a corresponding entry to `CHANGES.md` :)

**Comment by cpitclaudel** [comment]
All done!

**Pull request merged**